File: Data Structures & Algorithms/three-integer-sum/submission-2.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        vector<vector<int>> res;
        for(int i = 0; i < nums.size() - 2; i++){
            if(nums[i] > 0) break;
            if(i > 0 && nums[i] == nums[i - 1]) continue;
            int j = i + 1;
            int k = nums.size() - 1;
            while(j < k && j < nums.size()){
                int target = nums[i] * -1;
                if(nums[j] + nums[k] == target){
                    res.push_back({nums[i], nums[j], nums[k]});
                    j++;
                    k--;
                    while(j < k && nums[j] == nums[j - 1]) j++;
                }
                else if(nums[j] + nums[k] > target) k--;
                else j++;
            }
        }
        return res;
    }
};

File: Data Structures & Algorithms/car-fleet/submission-2.py
class Solution:
    def carFleet(self, target: int, position: List[int], speed: List[int]) -> int:
        carDets = []
        for i in range(len(position)):
            carDets.append([position[i], speed[i]])
        
        carDets.sort(reverse=True)

        order = []
        for data in carDets:
            time = (float(target - data[0]) / data[1])
            while len(order) > 0 and time <= order[-1]:
                order.pop()

            order.append(time)
        return len(order)